### ISSUE TITLE
Parse action doesn't validate commit hash

### ISSUE THREAD

**Issue by cassidyjames** [elementary/appcenter-reviews#68]
## What Happened

An app dev appears to have rewritten a release tag; https://github.com/elementary/appcenter-reviews/pull/66. The parse and build actions happily succeeded even though the listed commit doesn't match the release's commit.

## Expected Behavior

The parse action should check the commit of the release and fail if it doesn't match.

**Comment by danrabbit** [comment]
It seems like the dev rewrote history a bit and tagged their release to another commit after it was submitted

**Comment by cassidyjames** [comment]
Correct. But further runs of the parse action should have caught that, e.g. when merging main into the PR.

**Comment by btkostner** [comment]
Both the parse and manifest actions do nothing with git. That is handled by https://github.com/elementary/appcenter-reviews/blob/main/.github/workflows/review.yml#L68. We just need some bash-fu to fail if it's not correct here.
